docs/wiki/The CI environment - Deployment validation.md

```diff
@@ -0,0 +1,140 @@
+
+The parameter files used in this stage should ideally cover as many configurations as possible to validate the template flexibility, i.e. to verify that the module can cover multiple scenarios in which the same Azure resource may be used. Using the example of a CosmosDB module we may want to have one parameter file for the minimum amount of required parameters, one parameter file for each CosmosDB type to test individual configurations and at least one parameter file testing the supported extension resources such as RBAC & diagnostic settings.
+
+> **Note**: Since every customer environment might be different due to applied Azure Policies or security policies, modules might behave differently and naming conventions need to be verified beforehand.
+
+### Output example
+
+<img src=".\media\CIEnvironment\deploymentValidationOutput.png" alt="Deployment Validation Output" height="400">
+
+## Resource removal
+
+This paragraph describes how the removal of resources deployed by a module is performed and how to modify the default behavior if a specific module or resource type needs it.
+
+The removal step is triggered after the deployment completes. It takes care of removing all resources deployed as part of the previous deployment step. The reason is twofold:
+
+- Make sure to keep the validation subscription cost as low as possible.
+- Allow test deployments from scratch at every run.
+
+However, the removal step can be skipped in case further investigation on the deployed resource is needed. This can be controlled when running the module pipeline leveraging [Module pipeline inputs](./The%20CI%20environment%20-%20Pipeline%20design#module#module-pipeline-inputs).
+
+### How it works
+
+The removal process will remove all resources created during deployment. The list is identified by:
+
+1. Recursively fetching the list of resource IDs created through your deployment (resources created by deployments created by the parent one will be fetched too).
+1. Ordering the list based on resource IDs segment count (ensures child resources are removed first. E.g. `storageAccount/blobServices` comes before `storageAccount` as it has one more segments delimited by `/`).
+1. Filtering out from the list any resource used as dependencies for different modules (e.g. the commonly used Log Analytics workspace).
+1. Moving specific resource types to the top of the list (if a certain order is required). For example `vWAN` requires its `Virtual Hubs` to be removed first, even though they are no child-resources.
+
+After a resource is removed (this happens after each resource in the list), the script will execute, if defined, a **post removal operation**. This can be used for those resource types that requires a post processing, like purging a soft-deleted key vault.
```

Comment:
```suggestion
After a resource is removed (this happens after each resource in the list), the script will execute, if defined, a **post removal operation**. This can be used for those resource types that requires a post-processing, like purging a soft-deleted key vault.
```